### PR TITLE
Added support for setting service latency

### PR DESCRIPTION
**Pull request by aricart**

### CODE REVIEW

**Comment by derekcollison** [comment]
For those not too familiar with the low level code base a UX example in the PR description would be helpful.

File: cmd/addexport.go [hunk 1]
```diff
@@ -44,6 +45,8 @@ func createAddExportCmd() *cobra.Command {
 	cmd.Flags().StringVarP(&params.subject, "subject", "s", "", "subject")
 	cmd.Flags().BoolVarP(&params.service, "service", "r", false, "export type service")
 	cmd.Flags().BoolVarP(&params.private, "private", "p", false, "private export - requires an activation to access")
+	cmd.Flags().StringVarP(&params.latSubject, "lat-report", "", "", "latency report subject (services only)")
```
Comment (derekcollison) [inline cmd/addexport.go]: Would just do `latency` here.

File: cmd/addexport.go [hunk 2]
```diff
@@ -44,6 +45,8 @@ func createAddExportCmd() *cobra.Command {
...
+	cmd.Flags().IntVarP(&params.latSampling, "lat-freq", "", 0, "latency report frequency [1-100] (services only)")
```
Comment (derekcollison) [inline cmd/addexport.go]: Suggest sampling to be consistent.

File: cmd/addexport.go [hunk 3]
```diff
@@ -133,6 +143,40 @@ func (p *AddExportParams) PreInteractive(ctx ActionCtx) error {
 		return err
 	}
 
+	if p.export.IsService() {
+		ok, err := cli.PromptBoolean("track service latency", false)
+		if err != nil {
+			return err
+		}
+		if ok {
+			_, err = cli.Prompt("sampling frequency percentage [1-100]", "", false, func(s string) error {
```
Comment (derekcollison) [inline cmd/addexport.go]: Just `sampling percentage` IMO

File: cmd/addexport.go [hunk 4]
```diff
@@ -133,6 +143,40 @@ func (p *AddExportParams) PreInteractive(ctx ActionCtx) error {
...
+				v, err := strconv.Atoi(s)
+				if err != nil {
+					return err
+				}
+				if v < 1 || v > 100 {
```
Comment (derekcollison) [inline cmd/addexport.go]: If you wanted to here you could accept 0 to turn off. Not critical but you and I discussed.

File: cmd/addexport.go [hunk 5]
```diff
@@ -133,6 +143,40 @@ func (p *AddExportParams) PreInteractive(ctx ActionCtx) error {
...
+					return errors.New("sampling must be between 1 and 100 inclusive")
+				}
+				p.latSampling = v
+				return nil
+			})
+			p.latSubject, err = cli.Prompt("subject to send latency information", "", true, func(s string) error {
```
Comment (derekcollison) [inline cmd/addexport.go]: We have been referring to them as latency metrics.

Comment (derekcollison) [reply cmd/addexport.go]: Or latency result metrics.

File: cmd/describer.go [hunk 6]
```diff
@@ -156,13 +156,21 @@ func (e *ExportsDescriber) Describe() string {
 	table.UTF8Box()
 
 	table.AddTitle("Exports")
-	table.AddHeaders("Name", "Type", "Subject", "Public", "Revocations")
+	table.AddHeaders("Name", "Type", "Subject", "Public", "Revocations", "Monitoring")
```
Comment (derekcollison) [inline cmd/describer.go]: Maybe Tracking, again just trying to be as consistent as we can between programmatic, config, JWT and this tool.

File: go.sum [hunk 7]
```diff
@@ -78,6 +78,12 @@ github.com/nats-io/jwt v0.2.12 h1:Y3YLoJey+Q/yMk/1Ig3xhWxYXE7vNSefozkArIcnSlU=
 github.com/nats-io/jwt v0.2.12/go.mod h1:mQxQ0uHQ9FhEVPIcTSKwx2lqZEpXWWcCgA7R6NrWvvY=
 github.com/nats-io/jwt v0.2.13-0.20190726194050-829b612a49c3 h1:5vVSRJhjWOTv/TeJX1NUGuDYbZkfcWTu/97AHlsC02o=
 github.com/nats-io/jwt v0.2.13-0.20190726194050-829b612a49c3/go.mod h1:mQxQ0uHQ9FhEVPIcTSKwx2lqZEpXWWcCgA7R6NrWvvY=
+github.com/nats-io/jwt v0.2.16 h1:7lZ34jS9YAnW3gg/CITJuxjZnSI69kx/rgQAW4ro7G8=
```
Comment (derekcollison) [inline go.sum]: `go mod tidy`

**Comment by derekcollison** [comment]
I think you should have waited for more feedback. Let's do another PR to address the changes I requested.

File: cmd/addexport.go (hunk 4 above)
Comment (aricart) [reply cmd/addexport.go]: ok - let's do that.

**Pull request merged**